conventionalcommits/type.go

```diff
@@ -16,30 +16,93 @@ const (
 )
 
 var (
-	choreTypes   = []string{"build", "chore", "ci", "docs", "style", "refactor", "perf", "test"}
-	fixTypes     = []string{"fix"}
-	featureTypes = []string{"feat"}
-	allTypes     []string
+	defaultChoreTypes   = []string{"build", "chore", "ci", "docs", "style", "refactor", "perf", "test"}
+	defaultFixTypes     = []string{"fix"}
+	defaultFeatureTypes = []string{"feat"}
 )
 
-func initType() {
-	for _, types := range [][]string{choreTypes, fixTypes, featureTypes} {
-		allTypes = append(allTypes, types...)
+type TypeClassifier struct {
+	choreTypes   []string
+	fixTypes     []string
+	featureTypes []string
+	allTypes     []string
+}
+
+func NewTypeClassifier() *TypeClassifier {
+	tc := &TypeClassifier{
+		choreTypes:   append([]string{}, defaultChoreTypes...),
+		fixTypes:     append([]string{}, defaultFixTypes...),
+		featureTypes: append([]string{}, defaultFeatureTypes...),
 	}
+	tc.updateAllTypes()
+	return tc
 }
 
-func StringToType(s string) (Type, error) {
-	if slices.Contains(choreTypes, strings.ToLower(s)) {
-		return Chore, nil
+func NewTypeClassifierWithCustomPrefixes(customChoreTypes, customFixTypes, customFeatureTypes []string) *TypeClassifier {
+	tc := &TypeClassifier{}
+	
+	if len(customChoreTypes) > 0 {
+		tc.choreTypes = customChoreTypes
+	} else {
+		tc.choreTypes = append([]string{}, defaultChoreTypes...)
+	}
+	
+	if len(customFixTypes) > 0 {
+		tc.fixTypes = customFixTypes
+	} else {
+		tc.fixTypes = append([]string{}, defaultFixTypes...)
 	}
+	
+	if len(customFeatureTypes) > 0 {
+		tc.featureTypes = customFeatureTypes
+	} else {
+		tc.featureTypes = append([]string{}, defaultFeatureTypes...)
+	}
+	
+	tc.updateAllTypes()
+	return tc
+}
 
-	if slices.Contains(fixTypes, strings.ToLower(s)) {
-		return Fix, nil
+func (tc *TypeClassifier) updateAllTypes() {
+	tc.allTypes = nil
+	for _, types := range [][]string{tc.choreTypes, tc.fixTypes, tc.featureTypes} {
+		tc.allTypes = append(tc.allTypes, types...)
 	}
+}
 
-	if slices.Contains(featureTypes, strings.ToLower(s)) {
+func (tc *TypeClassifier) GetAllTypes() []string {
+	return tc.allTypes
```

Comment:
Do we actually need the field `.allTypes`? Instead of updating it at various places (which is error-prone), it may be better to just dynamically create this list here. It's not performance-critical, so this shouldn't be an issue.